ocaml/xapi/xapi_ha.ml

```diff
@@ -1123,7 +1133,14 @@ let ha_set_excluded __context _localhost =
   ()
 
 let ha_stop_daemon __context _localhost =
+  let module Fw =
+    ( val Firewall.firewall_provider !Xapi_globs.firewall_backend
+        : Firewall.FIREWALL
+      )
+  in
+  Fw.update_firewall_status ~service:Firewall.Xenha ~status:Firewall.Disabled ;
   Monitor.stop () ;
```

Comment:
Shouldn't the monitoring stop before the firewall is disabled? you might get errors otherwise. Also decide whether to disable the firewall anyway in the case the stop fails with an exception